### PR TITLE
Adds `entropy` of coupling computations and instantiates `Dotp` to `LRCGeometry`

### PR DESCRIPTION
**Pull request by marcocuturi**
computing the entropy of a coupling solution can be done in two different ways:
- `entropy_proxy`: this computation is only valid when the Sinkhorn algorithm has converged to a reasonable tolerance, as it exploits primal/dual equality of objectives. Used by default.
- `entropy_exact` materializes the coupling and computes its entropy. This is a lot more compute heavy, but more reliable.

`proxy` is used by default.

also, `Dotp` can now be instantiated as `LRCGeometry` quite trivially.

also, introduction of `normalized_entropy` for couplings on assignment problems, leveraging the fact that entropy of a bistochastic assignemnt matrix is sandwiched between `log n` and `2 logn`

### CODE REVIEW

File: tests/geometry/lr_cost_test.py [hunk 1]
```diff
@@ -72,6 +74,28 @@ def test_conversion_pointcloud(
...
+      rng: jax.Array,
+  ):
+    """Test conversion from PointCloud to LRCGeometry."""
+    n, m, d = 17, 11, 3
+    rngs = jax.random.split(rng, 3)
+    x = jax.random.normal(rngs[0], (n, d)) + .1
+    y = jax.random.normal(rngs[1], (m, d))
+
+    geom = pointcloud.PointCloud(x, y, cost_fn=costs.Dotp())
+    geom_lr = geom.to_LRCGeometry()
```
Comment (michalk8) [inline tests/geometry/lr_cost_test.py:88]: Please add `isinstance(geom_lr, low_rank.LRCGeometry)` just to be sure.

File: tests/solvers/linear/sinkhorn_test.py [hunk 2]
```diff
@@ -601,6 +602,33 @@ def test_primal_cost_pointcloud(self, cost_fn):
     cost = jnp.sum(out.matrix * out.geom.cost_matrix)
     np.testing.assert_allclose(cost, out.primal_cost, rtol=1e-5, atol=1e-5)
 
+  @pytest.mark.fast.with_args(cost_fn=[costs.SqEuclidean(), costs.Dotp()],)
```
Comment (michalk8) [inline tests/solvers/linear/sinkhorn_test.py:605]: The extra `,` can be removed.

File: tests/solvers/linear/sinkhorn_test.py [hunk 3]
```diff
@@ -601,6 +602,33 @@ def test_primal_cost_pointcloud(self, cost_fn):
...
+    out = solver(lin_prob)
+    ent_transport = jnp.sum(jsp.special.entr(out.matrix))
+    np.testing.assert_allclose(
+        ent_transport, out.entropy_proxy, atol=1e-3, rtol=1e-3
+    )
+    np.testing.assert_allclose(
+        ent_transport, out.entropy_exact, atol=1e-3, rtol=1e-3
+    )
+
+  @pytest.mark.fast.with_args(cost_fn=[costs.SqEuclidean(), costs.Dotp()],)
```
Comment (michalk8) [inline tests/solvers/linear/sinkhorn_test.py:621]: Same here, let's remove the extra `,`.

File: src/ott/solvers/linear/sinkhorn.py [hunk 4]
```diff
@@ -492,6 +492,34 @@ def g(self) -> jnp.ndarray:
...
+    coincide). As a result, it does not require materializing the transport
+    matrix, just applying it to the cost itself, which can be significantly
+    cheaper when the cost matrix admits a low-rank factorization.
+    """
+    assert self.ot_prob.is_balanced, "Entropy only valid for balanced problems."
+    return (self.primal_cost - self.ent_reg_cost) / self.geom.epsilon
+
+  @property
+  def entropy_exact(self) -> jnp.ndarray:
+    """Eaxct entropy of the coupling when the problem is balanced.
```
Comment (michalk8) [inline src/ott/solvers/linear/sinkhorn.py]: Typo in `Eaxct`.

File: tests/geometry/lr_cost_test.py (hunk 1 above)
Comment (marcocuturi) [reply tests/geometry/lr_cost_test.py:88]: good catch

**Review by michalk8 — APPROVED** [review]
LGTM, thanks!
